File: Algorithms/Linear-search/LinearSearch.cpp
#include <bits/stdc++.h> 
using namespace std; 

int linearSearch(int a[], int n, int x) 
{ 
	int i; 
	for (i = 0; i < n; i++) 
		if (a[i] == x) 
			return i; 
	return -1; 
} 

// Driver code 
int main() 
{ 
	int n;
	cout<<"ENTER SIZE OF ARRAY:- ";
	cin>>n;
	int a[n];
	cout<<"ENTER ARRAY ELEMENTS:- ";
	for(int i=0;i<n;i++)
	cin>>a[i];
	
	cout<<"ARRAY ELEMENTS ARE :- ";
	for(int i=0;i<n;i++)
	cout<<a[i]<<" ";
	
	int x;
	cout<<"\nENTER ELEMENT TO BE SEARCHED:- ";
	cin>>x;
	int index = linearSearch(a,n,x); 
	if (index == -1) 
		cout << "ELEMENT IS NOT PRESENT IN ARRAY!\n"; 
	else
		cout << "ELEMENT IS PRESENT AT POSITION :-  " << index; 

	return 0; 
} 

File: Algorithms/Linear-search/Linear-Search.java
import java.util.Scanner;


/**
 * 
 * 
 * Linear Search scans each elements in the array 
 * 
 * Time Complexity:-
 * 		Best Case:- element present at the start of the element 
 * 					O(1)
 * 		
 * 		Worst Case :- element not present in the list
 * 					O(n)
 *
 * Space Complexity:-  O(1) Constant Space
 */
public class LinearSearch {

	public static int search(int[] arr, int key) {

		for(int i=0;i<arr.length;i++)
		{
			if(arr[i] == key)
				return i;
		}
		
		return -1;
	}
}


class LinearSearchImplementation {
	
	public static void main(String[] args) {

		Scanner s=new Scanner(System.in);
		System.out.println("Enter the number of Elements");
		
		int n=s.nextInt();
		int arr[]=new int[n];
		
		System.out.println("Enter the Elements");
		for(int i=0;i<n;i++)
			arr[i]=s.nextInt();
		
		System.out.print("Enter the Element to be searched :");
		int key=s.nextInt();
	
		int result = LinearSearch.search(arr,key);
		
		if(result == -1)
			System.out.println("The element is not present in list");
		else
			System.out.print("The element is present at :"+ (result+1)+" position" );

	}
	
}

/*
 * 
Enter the number of Elements
5
Enter the Elements
1 2 10 9 3
Enter the Element to be searched :10
The element is present at :3 position


Enter the number of Elements
5
Enter the Elements
1 2 10 9 3
Enter the Element to be searched :7
The element is not present in list






*/

